pkg/controller/account/byoc.go

```diff
@@ -387,3 +362,29 @@ func validateBYOCClaim(accountClaim *awsv1alpha1.AccountClaim) error {
 
 	return nil
 }
+
+// replaceSecretAccessKey will create a new access key and use it to replace a users current one
+func replaceSecretAccessKey(reqLogger logr.Logger, client awsclient.Client, secret corev1.Secret, userName string, accessKeyID *string) (corev1.Secret, error) {
+	// Create new BYOC access keys
+	userSecretInfo, err := CreateUserAccessKey(reqLogger, client, userName)
+	if err != nil {
+		failedToCreateUserAccessKeyMsg := fmt.Sprintf("Failed to create IAM access key for %s", userName)
+		reqLogger.Info(failedToCreateUserAccessKeyMsg)
+		return secret, err
+	}
+
+	// Delete original BYOC access key
+	_, err = client.DeleteAccessKey(&iam.DeleteAccessKeyInput{AccessKeyId: accessKeyID})
+	if err != nil {
+		reqLogger.Error(err, "Failed to delete BYOC access keys")
+		return secret, err
+	}
+
+	// Update secret
+	secret.Data =
+		map[string][]byte{
+			"aws_access_key_id":     []byte(*userSecretInfo.AccessKey.AccessKeyId),
+			"aws_secret_access_key": []byte(*userSecretInfo.AccessKey.SecretAccessKey),
+		}
+	return secret, nil
```

Comment:
I'm wondering if we want to return a secret here, or if we should keep the part about the secret in the function calling this one and just return the AccessKeyID and SecretAccessKey and an error here instead?  If we were pulling data out of the secret at first to be able to rotate the keys then I'd have different feelings about this, but I feel the part about putting the data into the secret just smells wrong here.  I'm not dying on this hill, though.